### PR TITLE
[opt](nereids)do not apply CSE(Common Sub Expression) upon multiDataSink

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -1841,7 +1841,7 @@ public PlanFragment visitPhysicalProject(PhysicalProject<? extends Plan> project
         List<Expr> projectionExprs = null;
         List<Expr> allProjectionExprs = Lists.newArrayList();
         List<Slot> slots = null;
-        if (project.hasMultiLayerProjection()) {
+        if (project.hasMultiLayerProjection() && !(inputFragment instanceof MultiCastPlanFragment)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1844]: will we get wrong result if we just change logic in translator?
btw: add comment to explain why cannot do multi layer project

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1844]: when we add multiLayerProjection, we do not modify its original projects. That is why only translator need to be modified.

**Comment by englefly** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by englefly** [comment]
run external

**Review by starocean999 — APPROVED** [review]

**Pull request merged**